docs/Dialects/Interop/RationaleInterop.md

```diff
@@ -0,0 +1,590 @@
+    - [Instance-side Lowering](#instance-side-lowering)
+    - [Container-side Lowering](#container-side-lowering)
+    - [Bridging between Interop Mechanisms](#bridging-between-interop-mechanism)
+    - [How to use this dialect](#how-to-use-this-dialect)
+    - [Design considerations](#design-considerations)
+
+## Introduction
+
+The increasing number of CIRCT-based backends as well as the usage of many
+external tools raises questions about interoperability and composition of
+those tools. Are we supposed to write interop-layers between them by hand or
+can this also be automatically generated? Can we generate interfaces to the
+outside in multiple standardized formats/languages automatically?
+
+Let's pick simulation as a concrete example. We have Verilator, CIRCT-based
+simulators like the event-driven `llhd-sim`, and at some point maybe another
+non-event-driven simulator to compete with Verilator in terms of performance.
+They all provide their own interface for writing test-benches in a way that
+suits them best. Verilator has its custom C++ interface as well as SystemC
+support, `llhd-sim` expects the test-bench to be written in LLHD (e.g., compiled
+from SystemVerilog), and nobody knows what the interface of the third would look
+like (maybe a custom C-header?). This means, if we want to simulate a design
+with all three, we have to write the test-bench three times or at least some
+layer between the TB and the simulator interfaces. This dialect aims to solve
+this issue by auto-generating an interface from a selection of standarized
+formats (e.g., a SystemC or SystemVerilog module) such that the test-bench has
+to be written only once and switching to another simulator just means changing
+one CLI argument.
+What if we have to use some blackbox Verilog code in the design? Is it compiled
+by a (not yet existent) SystemVerilog front-end and simulated by the CIRCT-based
+simulators or could it also be useful to simulate the blackbox using Verilator
+and everything surrounding it using the CIRCT-based simulator? This dialect can
+provide the latter by implementing an interop operation and one lowering per
+simulator rather than custom pair-wise interop implementations.
+
+We make a clear distinction between procedural and structural interop because
+it is not possible to provide direct interop between a purely structural
+language and a purely procedural language. To achieve interop between them,
+one has to use at least one bridging language that has (some) support for both
+as a translation layer.
+
+## Procedural Interop
+
+The `interop` dialect provides a few operations to represent partially lowered
+procedural interop instances presented in the
+[next chapter](#representing-partial-lowerings) as well as an interface to be
+implemented by the surrounding module and rewrite patterns to call into those
+interface implementations presented in the chapter about
+['Container-side Lowering'](#container-side-lowering).
```

Comment:
Mega-nit: these are really "sections" and not "chapters" (unless this was compiled to a book). 😉